### PR TITLE
Like history functionality

### PR DESCRIPTION
**Pull request by zoef-yt**
In this pr, I added
- like functionality
- history functionality

[Check only me](https://github.com/zoef-yt/game-plex-react/pull/6/commits/43a895c6ae6cf7ef23c70f783c6e7a9d6d2f01ad)
[Preview link](https://deploy-preview-6--game-plex-zoef.netlify.app/)

### CODE REVIEW

**Review by iRohitGaur — APPROVED** [review]
UI looks great.

Nice work! Good to merge 👍🏻

File: src/Context/HistoryContext/HistoryContext.jsx [hunk 1]
```diff
@@ -4,27 +4,62 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
-	]);
-	return <HistoryContext.Provider value={{ historyList }}>{children}</HistoryContext.Provider>;
+	const [historyList, useHistoryList] = useState([]);
+
+	const { response: historyResponse, operation: historyServerCall, error: historyError } = useAxios();
+
+	const token = localStorage.getItem('token');
+
+	useEffect(() => {
+		if (historyResponse != null && historyResponse.history) {
```
Comment (DivyanshSareen) [inline src/Context/HistoryContext/HistoryContext.jsx:14]: use !== instead of !=

**Review by DivyanshSareen — APPROVED** [review]
Everything looks great! good to merge

File: src/Context/HistoryContext/HistoryContext.jsx (hunk 1 above)
Comment (muskanbagrecha) [inline src/Context/HistoryContext/HistoryContext.jsx:14]: Can use optional chaining for concise code.

File: src/Context/HistoryContext/HistoryContext.jsx [hunk 2]
```diff
@@ -4,27 +4,62 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
+			useHistoryList(historyResponse.history);
+		} else {
+			getHistory();
+		}
+	}, [historyResponse]);
+
+	useEffect(() => {
```
Comment (muskanbagrecha) [inline src/Context/HistoryContext/HistoryContext.jsx:21]: You can remove this.

File: src/Context/LikedVideosContext/LikedVideosContext.jsx [hunk 3]
```diff
@@ -4,11 +4,49 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
-		{ _id: 'SzuoB4_RpT4', title: "Marvel's Avengers Review(*NO SPOLIERS*)", creator: 'Zoef Games', categoryName: 'Podcast', id: '2' },
-	]);
-	return <LikesContext.Provider value={{ likesList }}>{children}</LikesContext.Provider>;
+	const [likesList, useLikesList] = useState([]);
+	const { response: likeVideosResponse, operation: likeVideosServerCall, loading: loadingWatchLater, error: likeVideosError } = useAxios();
+
+	const token = localStorage.getItem('token');
+
+	useEffect(() => {
+		if (likeVideosResponse != null && likeVideosResponse.likes) {
```
Comment (muskanbagrecha) [inline src/Context/LikedVideosContext/LikedVideosContext.jsx:13]: Can use optional chaining for concise code.

File: src/Context/HistoryContext/HistoryContext.jsx [hunk 4]
```diff
@@ -4,27 +4,62 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
-		},
-		{
-			_id: 'SzuoB4_RpT4',
-			title: "Marvel's Avengers Review(*NO SPOLIERS*)",
-			creator: 'Zoef Games',
-			categoryName: 'Podcast',
-		},
-	]);
-	return <HistoryContext.Provider value={{ historyList }}>{children}</HistoryContext.Provider>;
+	const [historyList, useHistoryList] = useState([]);
```
Comment (muskanbagrecha) [inline src/Context/HistoryContext/HistoryContext.jsx:7]: Personal opinion: I dont like how you are prefixing the setter function with "use". It's confusing.

File: src/Context/LikedVideosContext/LikedVideosContext.jsx [hunk 5]
```diff
@@ -4,11 +4,49 @@ import { useAxios } from '../../CustomHooks/useAxios';
 const LikesContext = createContext();
 
 const LikesProvider = ({ children }) => {
-	const [likesList, useLikesList] = useState([
-		{ _id: 'O-w-NKMowIY', title: 'Valorant Skin Challenge | Zoef Games', creator: 'Zoef Games', categoryName: 'Challenges', id: '1' },
-		{ _id: 'SzuoB4_RpT4', title: "Marvel's Avengers Review(*NO SPOLIERS*)", creator: 'Zoef Games', categoryName: 'Podcast', id: '2' },
-	]);
-	return <LikesContext.Provider value={{ likesList }}>{children}</LikesContext.Provider>;
+	const [likesList, useLikesList] = useState([]);
```
Comment (muskanbagrecha) [inline src/Context/LikedVideosContext/LikedVideosContext.jsx:7]: Even here.

**Review by muskanbagrecha — APPROVED** [review]
Everything looks good. Clean code. Do look into the suggestions.

File: src/Context/HistoryContext/HistoryContext.jsx (hunk 1 above)
Comment (zoef-yt) [reply src/Context/HistoryContext/HistoryContext.jsx:14]: Done.

File: src/Context/LikedVideosContext/LikedVideosContext.jsx (hunk 3 above)
Comment (zoef-yt) [reply src/Context/LikedVideosContext/LikedVideosContext.jsx:13]: done.

File: src/Context/HistoryContext/HistoryContext.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Context/HistoryContext/HistoryContext.jsx:21]: done.

File: src/Context/HistoryContext/HistoryContext.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Context/HistoryContext/HistoryContext.jsx:7]: Changed was a mistake.

File: src/Context/LikedVideosContext/LikedVideosContext.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/Context/LikedVideosContext/LikedVideosContext.jsx:7]: done.

File: src/Context/HistoryContext/HistoryContext.jsx (hunk 1 above)
Comment (zoef-yt) [reply src/Context/HistoryContext/HistoryContext.jsx:14]: done

**Comment by zoef-yt** [comment]
@muskanbagrecha done with all the suggested changes in pr #8

**Pull request merged**